source/frontend/src/features/documents/DocumentPreviewContainer.tsx

```diff
@@ -0,0 +1,167 @@
+> = ({ View, createFileUrl }) => {
+  const { showDocumentPreview, previewDocumentId, setShowDocumentPreview, setPreviewDocumentId } =
+    useContext(DocumentViewerContext);
+  const { setModalContent, setDisplayModal } = useContext(ModalContext);
+  const {
+    retrieveDocumentDetail,
+    downloadDocumentFilePageImage,
+    getDocumentFilePageList,
+    downloadWrappedDocumentFileLatest,
+    retrieveDocumentDetailLoading,
+    getDocumentFilePageListLoading,
+  } = useDocumentProvider();
+  const [documentPages, setDocumentPages] = useState<LoadedPages[]>([]);
+  const [currentPage, setCurrentPage] = useState<number>(0);
+
+  const loadDocumentPageImage = useCallback(
+    async (documentId: number, documentFileId: number, documentFilePageId: number) => {
+      //todo: handle image failure
+      return (createFileUrl ? createFileUrl : URL.createObjectURL)(
+        await downloadDocumentFilePageImage(documentId, documentFileId, documentFilePageId),
+      );
+    },
+    [downloadDocumentFilePageImage, createFileUrl],
+  );
+
+  const resetDocumentPreview = useCallback(() => {
+    setShowDocumentPreview(false);
+    setPreviewDocumentId(null);
+    setCurrentPage(0);
+    setDocumentPages([]);
+  }, [setPreviewDocumentId, setShowDocumentPreview]);
+
+  const loadPagesForDocumentId = useCallback(async () => {
+    const data = await retrieveDocumentDetail(previewDocumentId);
+    if (
+      data?.payload?.datetime_created &&
+      moment(data.payload.datetime_created).add(1, 'minutes').isAfter(moment())
+    ) {
+      setModalContent({
+        title: 'Document Still Processing',
+        message:
+          'This document has been uploaded very recently, please wait a few minutes before trying to preview it. You may download the document in the meantime (using the "eye" icon).',
+        variant: 'info',
+        handleOk: () => {
+          setDisplayModal(false);
+          resetDocumentPreview();
+        },
+      });
+      setDisplayModal(true);
+    } else if (
+      data?.httpStatusCode !== ApiGen_System_HttpStatusCode.OK ||
+      !data?.payload?.file_latest
+    ) {
+      setModalContent({
+        title: 'Document Preview Error',
+        message:
+          'Failed to retrieve document details. This may indicate the document is still being processed, or failed to upload. If the issue persists, please contact the system administrator.',
+        variant: 'error',
+        handleOk: () => {
+          setDisplayModal(false);
+          resetDocumentPreview();
+        },
+      });
+      setDisplayModal(true);
+    } else if (data?.payload?.file_latest) {
+      const pages = await getDocumentFilePageList(previewDocumentId, data.payload.file_latest.id);
+
+      //todo: handle various doc failures.
+      if (pages?.length > 0) {
+        const documentPageImage = await loadDocumentPageImage(
+          previewDocumentId,
+          data.payload.file_latest.id,
+          pages[0].id,
```

Comment:
This looks like is only necessary have the first page ready after the pages have been retrieved. Is this necessary because the useEffect at line 133 wont load the first page? If so, please add a comment with a small blurb indicating that. Otherwise, I would expect that loading the first page would work the same as loading all other pages

Reply:
testing to see if this is necessary, this may be an artifact of a previous implementation.